src/utils/buildSearchQueryMetadata.ts

```diff
@@ -0,0 +1,24 @@
+import { TSearchCriteria, TThemeConfig } from "@types";
+
+export const buildSearchQueryMetadata = (
+  query: TSearchCriteria,
+  metadataValue: string | string[],
+  taxonomyKey: string,
+  themeConfig: TThemeConfig
+) => {
+  let metadataForApi: string[];
+  const configMetadata = themeConfig.filters.find((f) => f.taxonomyKey === taxonomyKey);
+  if (configMetadata) {
+    // remove existing metadata filters from the metadata
+    query.metadata = query.metadata.filter((m) => m.name !== configMetadata.apiMetaDataKey);
+    if (Array.isArray(metadataValue)) {
+      metadataForApi = metadataValue;
+    } else {
+      metadataForApi = [metadataValue];
+    }
+    metadataForApi.map((m) => {
+      query.metadata.push({ name: configMetadata.apiMetaDataKey, value: decodeURI(m) });
+    });
+  }
+  return null;
```

Comment:
Suggestion: This function should return a new `query` object rather than mutating the function parameter. This feels like a side effect rather than functional programming, and is much harder to test.

Reply:
Good point, I'll look at that